typed/reconcile_schema.go

```diff
@@ -231,21 +226,16 @@ func (v *reconcileWithSchemaWalker) visitMapItems(t *schema.Map, element *fieldp
 }
 
 func (v *reconcileWithSchemaWalker) doMap(t *schema.Map) (errs ValidationErrors) {
-	// reconcile maps and structs changed from granular to atomic
+	// reconcile maps and structs changed from granular to atomic.
+	// Note that migrations from atomic to granular are unsuported and will
+	// be treated as if they were always granular.
 	if !v.isAtomic && t.ElementRelationship == schema.Atomic {
 		if v.fieldSet != nil && v.fieldSet.Size() > 0 {
 			v.toRemove = fieldpath.NewSet(v.path) // remove all root and all children fields
 			v.toAdd = fieldpath.NewSet(v.path)    // add the root of the atomic
 		}
 		return errs
 	}
-	// reconcile maps changed from atomic to granular
-	if v.isAtomic && (t.ElementRelationship == schema.Separable || t.ElementRelationship == "") {
-		v.toAdd, errs = buildGranularFieldSet(v.path, v.value)
```

Comment:
Is `buildGranularFieldSet` still used anywhere?

Reply:
nope, good catch